modules/dataplex-autodq/variables.tf

```diff
@@ -0,0 +1,166 @@
+  type = object({
+    entity   = optional(string)
+    resource = optional(string)
+  })
+  validation {
+    condition     = length([for k, v in var.data : v if contains(["resource", "entity"], k) && v != null]) == 1
+    error_message = "Datascan data must specify one of 'entity', 'resource'."
+  }
+}
+
+variable "execution_schedule" {
+  description = "Schedule DataScan to run periodically based on a cron schedule expression. If not specified, the DataScan is created with `on_demand` schedule, which means it will not run until the user calls `dataScans.run` API."
+  type        = string
+  default     = null
+}
+
+variable "group_iam" {
+  description = "Authoritative IAM binding for organization groups, in {GROUP_EMAIL => [ROLES]} format. Group emails need to be static. Can be used in combination with the `iam` variable."
+  type        = map(list(string))
+  default     = {}
+  nullable    = false
+}
+
+variable "iam" {
+  description = "Dataplex AutoDQ  IAM bindings in {ROLE => [MEMBERS]} format."
+  type        = map(list(string))
+  default     = {}
+  nullable    = false
+}
+
+variable "iam_additive" {
+  description = "IAM additive bindings in {ROLE => [MEMBERS]} format."
+  type        = map(list(string))
+  default     = {}
+  nullable    = false
+}
+
+variable "iam_additive_members" {
+  description = "IAM additive bindings in {MEMBERS => [ROLE]} format. This might break if members are dynamic values."
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "iam_policy" {
+  description = "IAM authoritative policy in {ROLE => [MEMBERS]} format. Roles and members not explicitly listed will be cleared, use with extreme caution."
+  type        = map(list(string))
+  default     = null
+}
+
+variable "incremental_field" {
+  description = "The unnested field (of type Date or Timestamp) that contains values which monotonically increase over time. If not specified, a data scan will run for all data in the table."
+  type        = string
+  default     = null
+}
+
+variable "labels" {
+  description = "Resource labels."
+  type        = map(string)
+  default     = {}
+  nullable    = false
+}
+
+variable "name" {
+  description = "Name of Dataplex AutoDQ Scan."
+  type        = string
+}
+
+variable "prefix" {
+  description = "Optional prefix used to generate Dataplex AutoDQ DataScan ID."
+  type        = string
+  default     = null
+}
+
+variable "project_id" {
+  description = "The ID of the project where the Dataplex AutoDQ Scans will be created."
+  type        = string
+}
+
+variable "region" {
+  description = "Region for the Dataplex AutoDQ Scan."
+  type        = string
+}
+
+variable "row_filter" {
+  description = "A filter applied to all rows in a single DataScan job. The filter needs to be a valid SQL expression for a WHERE clause in BigQuery standard SQL syntax. Example: col1 >= 0 AND col2 < 10."
+  type        = string
+  default     = null
+}
+
+variable "rules" {
```

Comment:
Would it makes sense to explode some of the nested values as shown below?

```hcl
list(object({
    column               = optional(string)
    ignore_null          = optional(bool)
    dimension            = string
    threshold            = optional(number)
    non_null_expectation = optional(object({}))
    range_expectation = optional(object({
      min_value          = optional(number)
      max_value          = optional(number)
      strict_min_enabled = optional(bool)
      strict_max_enabled = optional(bool)
    }))
    regex_expectation = optional(string)
    set_expectation = optional(list(string))
    uniqueness_expectation = optional(object({}))
    statistic_range_expectation = optional(object({
      statistic          = string
      min_value          = optional(number)
      max_value          = optional(number)
      strict_min_enabled = optional(bool)
      strict_max_enabled = optional(bool)
    }))
    row_condition_expectation = optional(string)
    table_condition_expectation = optional(string)
  }))```

Reply:
Hey @juliocc, good point. My rationale for not simplifying this structure at the moment is to ensure compatibility / keep a 1:1 mapping for the rules configurations to the API protos and the input format for the gcloud argument `--data-quality-spec-file` in https://cloud.google.com/sdk/gcloud/reference/alpha/dataplex/datascan/create. One example I will add is how to pass in the rules as a yaml input file -- and we would like this yaml input file to be compatible with the REST API / gcloud. Hope that makes sense.

Reply:
Makes sense. I just mentioned this because there are places where we try to simplify the interface to make it easier to use the module (e.g. organization policies). 

You'll be the primary user of this so go with what you feel is right.